backend/lib/rpc-service/service.js

```diff
@@ -14,13 +15,18 @@ import { ClaimEvent, RawActorEvent, BlockEvent, RpcRespone } from './data-types.
  */
 export const rpcRequest = async (method, params) => {
   const reqBody = JSON.stringify({ method, params, id: 1, jsonrpc: '2.0' })
-  const response = await fetch(RPC_URL, {
-    method: 'POST',
-    headers: { 'content-type': 'application/json' },
-    body: reqBody
-  })
-
-  return Value.Parse(RpcRespone, (await response.json())).result
+  try {
+    const response = await pRetry(async () => {
+      return await fetch(RPC_URL, {
+        method: 'POST',
+        headers: { 'content-type': 'application/json' },
+        body: reqBody
+      })
+    }, { retries: 5 })
+    return Value.Parse(RpcRespone, await response.json()).result
+  } catch (e) {
+    console.error(`Failed to make RPC request: ${e.message}`)
+  }
```

Comment:
IIUC, when there is an error, this code prints the error to stderr and the function `rpcRequest` returns `undefined`. This means all callers of `rpcRequest` must understand this path and handle the `undefined` result.

I guess this works well in Rust, where you have `Result<T, E>` and the compiler tells you when you forget to handle the error path.

We don't have such guarantees in JavaScript. TypeScript _may_ be able to detect places where we don't check for `undefined`, but it's not reliable enough (we have `any` type in too many places), it's not enforced at runtime, and it's also difficult to spot during code reviews.

What we usually do: When we don't have a specific way how to handle the error, we either don't catch it or re-throw it with more details, letting the caller (or the caller of the caller) to deal with it.

Reply:
So in this particular case, I suggest the following code here, plus make sure we log unhandled errors in the top-level loop.

```suggestion
  } catch (e) {
    e.message = `Failed to make RPC request for ${method}: ${e.message}`
    throw e
  }
```

Also note that JavaScript errors include a stack trace that is often very useful for troubleshooting the issue, so it's best to log the entire error object (and let the console handle formatting for display), not only the error message. You cannot use a [template literal](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals) for that, you have to pass values to format as additional arguments to console methods.

```js
console.error('Failed to make RPC request:', e)
```

Reply:
I see you already have code in the top-level loop to report errors, all is good there 👍🏻 

https://github.com/filecoin-station/deal-observer/blob/492ff6aabbf10e38ee53a9c33e17bdeac83c1f06/backend/bin/deal-observer-backend.js#L38-L41